### PR TITLE
Updates to User Guide: 6/Dependencies

### PR DESCRIPTION
**Pull request by mwichmann**
Removed most references to md5 and tried to make the terminology used a bit more consistent.

Added some markup - refernced functions now hyperlinked on first mention of each section.

Doc-only change.

## Contributor Checklist:

* [ ] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [ ] I have updated `CHANGES.txt` (and read the `README.rst`)
* [X] I have updated the appropriate documentation

### CODE REVIEW

File: doc/user/depends.xml [hunk 1]
```diff
@@ -109,26 +109,26 @@ int main() { printf("Hello, world!\n"); }
     things when an input file changes,
     so that the built software is up to date.
     By default,
-    &SCons; keeps track of this through an
-    MD5 &signature;, or checksum, of the contents of each file,
+    &SCons; keeps track of this through a
+    <firstterm>content signature</firstterm>,
```
Comment (bdbaddog) [inline doc/user/depends.xml]: can we change this to use an entity?  I see was using &signature before?

File: doc/user/depends.xml [hunk 2]
```diff
@@ -1206,23 +1207,23 @@ Program('hello', 'hello.c', CPPPATH='.')
     <para>
 
     Apparently, the scanner does not know about the header dependency.
-    Being not a full-fledged C preprocessor, the scanner does not
+    Note being a full-fledged C preprocessor, the scanner does not
```
Comment (bdbaddog) [inline doc/user/depends.xml]: Not being vs Note being?

File: doc/user/depends.xml (hunk 1 above)
Comment (mwichmann) [reply doc/user/depends.xml]: maybe if *content signature* was the entity?

So, you made me look... right now we have two that don't exactly fit - and `<literal>` is not really the way those two should be marked up anyway, they're not programmatic elements like variables, functions, command lines, etc.
```
<!ENTITY signature "<literal xmlns='http://www.scons.org/dbxsd/v1.0'>signature</literal>">
<!ENTITY buildsignature "<literal xmlns='http://www.scons.org/dbxsd/v1.0'>build signature</literal>">
```

File: doc/user/depends.xml (hunk 2 above)
Comment (mwichmann) [reply doc/user/depends.xml]: Sigh. Fingers don't always type what brain thought about.

File: doc/user/depends.xml (hunk 1 above)
Comment (bdbaddog) [reply doc/user/depends.xml]: sounds reasonable? content signature as a new entity?

**Pull request merged**